### PR TITLE
Feat/new identities

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

**Review by theodelrieu — APPROVED** [review]

File: Identity/ProvisionalIdentity.cs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 using Newtonsoft.Json;
 using SnakeCaseStrategy = Newtonsoft.Json.Serialization.SnakeCaseNamingStrategy;
 using Tanker.Crypto;
-
+using System;
+using System.Text;
```
Comment (blastrock) [inline Identity/ProvisionalIdentity.cs:5]: You need these?

File: Identity/ProvisionalIdentity.cs [hunk 2]
```diff
@@ -29,12 +29,12 @@ public PublicProvisionalIdentity(SecretProvisionalIdentity secretIdentity)
             Target = secretIdentity.Target;
             Value = secretIdentity.Value;
             if (Target == "email") {
-                Target = "hashed_email";
                 var hashedEmail = CryptoCore.GenericHash(Encoding.UTF8.GetBytes(Value), 32);
                 Value = Convert.ToBase64String(hashedEmail);
             } else {
                 Value = CryptoCore.HashProvisionalValue(Value, secretIdentity.PrivateSignatureKey);
             }
+                Target = "hashed_" + Target;
```
Comment (blastrock) [inline Identity/ProvisionalIdentity.cs]: indentation

**Review by blastrock — APPROVED** [review]

File: Identity/ProvisionalIdentity.cs (hunk 2 above)
Comment (tux3) [reply Identity/ProvisionalIdentity.cs]: Fixed, thank you.

File: Identity/ProvisionalIdentity.cs (hunk 1 above)
Comment (tux3) [reply Identity/ProvisionalIdentity.cs:5]: System.Text for `Encoding.UTF8.GetByte`  
System for things like `ArgumentException`

**Pull request merged**